V8 style support

ryanhamley opened the pull request:
This PR fixes https://github.com/mapbox/mapbox-gl-language/issues/31 but in doing so, it drops support for all Mapbox styles based on vector tile source v7 as these styles are deprecated and Mapbox is encouraging all users to migrate to v8 styles. This change will require a major version release of this plugin. 

Additionally, this PR drops support for IE11 since Mapbox GL JS is also deprecating support for the browser.

This PR also updates the example pages to use the newest styles and current versions of Mapbox GL JS and the RTL plugin while adding new example pages for Italian, Simplified Chinese and Traditional Chinese.

The code now assumes that your `text-field` will include an expression of the form `['get', 'name_xx']`. This can be a top-level expression, in a `coalesce` expression or nested deeper in an expression. The plugin will also now support Italian (`it`), Simplified Chinese (`zh-Hans`) and Traditional Chinese (`zh-Hant`) by default. v7 style Chinese `zh` is no longer supported.

@mapbox/gl-js

samanpwbb reviewed (commented):
@ryanhamley this looks solid! I think we should get the docs updated to clarify requirements for using this library: 
- Streets v8
- Expression text values (no support for curly backets)
- Expects translated languages with coalesce expressions that fall back to non-null `name` field.

I see a few opportunities for improvement but I don't think these are necessary for merge:

1. There is a somewhat common edge case where user will have a text field that isn't already wrapped in a coalesce. Based on the logic implemented now, this could result in a lot of empty labels. To cover these cases, it'd be great if the plugin checked if a field values was wrapped in a coalesce, and added the coalesce if not.
2. To improve developer UX, add some validation to warn on curly bracket tokens, or warn on non-coalesce wrapped get expressions.
3. There are rare edge cases where language get expressions are used in _inputs_ to other expressions rather than outputs. Since the current implementation doesn't know about context when it applies transforms, there's no way to handle these cases. I would at least add a note about this in the readme. Take for example:

```js
["match", 
  ["get", "name"], 
  "Canada", 
  "America's Hat", 
  ["coalesce", 
    ["get", "name_en"], 
    ["get", "name"]
  ]
]
```

andrewharvey commented on index.js:
```diff
@@ -152,9 +153,12 @@ function changeLayerTextProperty(isLangField, layer, languageFieldName, excluded
+    var url = style.sources[sourceName].url;
+    // the source URL can reference the source version or the style version
+    // this check and the error forces users to migrate to styles using source version 8
+    return url.indexOf('mapbox.mapbox-streets-v8') > -1 || /mapbox-streets-v[1-9][1-9]/.test(url);
```
geojson sources don't have a url, so should we skip sources here without a url?

andrewharvey replied:
```suggestion
    return url && url.indexOf('mapbox.mapbox-streets-v8') > -1 || /mapbox-streets-v[1-9][1-9]/.test(url);
```

andrewharvey reviewed (approved):
Looks great. I've been running this in production with no issues noticed so far.

ryanhamley commented:
@samanpwbb I think I addressed all your concerns. This now handles the unwrapped case `text-field: ['get', 'name']` and provides a warning when token syntax is encountered and the documentation has been updated.

@andrewharvey That's great to hear that this is already working well! Once this is merged, I'm going to make a few updates and make a new release.

samanpwbb commented on README.md:
```diff
@@ -70,7 +70,18 @@ Create a new [Mapbox GL JS plugin](https://www.mapbox.com/blog/build-mapbox-gl-j
+["match", 
+  ["get", "name"], 
+  "Canada", 
+  "America's Hat", 
```
hehe maybe we should use an example that is less offensive to canadians. @tristen do you have suggestions.

tristen replied:
I likey. @willymaps?

ryanhamley replied:
```
["match", 
  ["get", "name"], 
  "California", 
  "Golden State", 
  ["coalesce", 
    ["get", "name_en"], 
    ["get", "name"]
  ]
]
```

samanpwbb reviewed (approved):
Looks good!
